Shazwazza/Articulate
# Routing windowslivewriter manifest & XMLRPC MetaWeblog to support OpenLiveWriter

* Adds the `wlwmanifest/{id}` route to return the XML used in conjunction with the RSD XML
* Gets the XMLRPC / MetaWeblog Controller for receiving HTTP POSTed XML for OpenLiveWriter
* Drops Cookcomputing.XMLRpc for `WilderMinds.MetaWeblog`
https://github.com/shawnwildermuth/MetaWeblog

## Help needed
I am unsure with DI to update the Root Articulate Blog Node ID - so appreciate any pointers @Shazwazza 
```csharp
private IPublishedContent BlogRoot()
{
    // TODO: Do not hardcode 1068 - need to pass it through from the controller route
    // When injecting this service into the controller - but how?!?
    var node = _umbracoContextAccessor.GetRequiredUmbracoContext().Content.GetById(1068);
    ...
}
```

## Testing
* Download Open Live Writer
* Have test site running & put a breakpoint in `MetaWeblogController` to help with ddebugging
* Add new configuration pointing to example site
* Add a new post - via theme detection
* It should delete the temp post & put it in the recycle bin
* Add a new post
  * Select existing category
  * Add a new category
  * Select existing tags
  * Add a new tag
  * Add one or more image (Use the silly effects in LiveWriter)
  * Save draft & verify
  * Publish & verify
* Close Open Live Writer
* File -> Open Recent Post -> Select blog account from list (Fetches recent posts)
  * Select an item
  * Use delete button - ensure blog post is moved to recycle bin
  * Select an item to edit
  * Update content, properties, categories & tags - verify changes are expected
